bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1201,8 +1201,8 @@ public void setEnabled(ThingUID thingUID, boolean enabled) {
 
         if (enabled) {
             // Enable a thing
-            if (thing.getStatus().equals(ThingStatus.ONLINE)) {
-                logger.debug("Thing {} is already in the required state.", thingUID);
+            if (thing.isEnabled()) {
```

Comment:
Actually this check is a lot better than before. But it will not prevent multiple initializations for sure. There will be still some milliseconds left before the `ThingStatus` will be set to `INITIALIZING`. The initialization process itself is covered by a [mutex lock](https://github.com/mdicke2s/openhab-core/blob/c318a96f065d68e937c37c1bed178ba1718ae019/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java#L632). We maybe have to introduce a similar lock for enabling / disabling process too.